scrape/scrape.go

```diff
@@ -574,20 +573,16 @@ func (sp *scrapePool) sync(targets []*Target) {
 // refreshTargetLimitErr returns an error that can be passed to the scrape loops
 // if the number of targets exceeds the configured limit.
 func (sp *scrapePool) refreshTargetLimitErr() error {
-	if sp.config == nil || sp.config.TargetLimit == 0 && !sp.targetLimitHit {
+	if sp.config == nil || sp.config.TargetLimit == 0 {
 		return nil
 	}
 	l := len(sp.activeTargets)
-	if l <= int(sp.config.TargetLimit) && !sp.targetLimitHit {
-		return nil
-	}
-	var err error
-	sp.targetLimitHit = l > int(sp.config.TargetLimit)
-	if sp.targetLimitHit {
+	targetLimitHit := l > int(sp.config.TargetLimit)
+	if targetLimitHit {
```

Comment:
```suggestion
	if l := len(sp.activeTargets); l > int(sp.config.TargetLimit) {
```

Reply:
Thanks for looking into this, applied your suggestion. :)